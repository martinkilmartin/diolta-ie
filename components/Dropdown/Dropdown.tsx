import { ReactNode } from 'react'

import clsx from 'clsx'

const ALIGN = {
  no: '',
  end: ' dropdown-end',
}

const OPEN_FROM = {
  bottom: '',
  top: ' dropdown-top',
  left: ' dropdown-left',
  right: ' dropdown-right',
}

const OPEN = {
  no: '',
  hover: ' dropdown-hover',
  open: ' dropdown-open',
}

type Props = {
  title?: string
  align?: keyof typeof ALIGN
  openFrom?: keyof typeof OPEN_FROM
  open?: keyof typeof OPEN
  children: ReactNode
}

const Dropdown = ({
  title = 'Change Theme',
  align = 'no',
  openFrom = 'bottom',
  open = 'no',
  children,
}: Props): JSX.Element => {
  return (
    <div
      className={clsx(
        'dropdown',
        ALIGN[align],
        OPEN_FROM[openFrom],
        OPEN[open]
      )}
    >
      <div tabIndex={0} className="m-1 btn btn-secondary">
        {title}
      </div>
      {children}
    </div>
  )
}

export default Dropdown
