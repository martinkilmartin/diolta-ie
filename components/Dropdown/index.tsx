export { default as Dropdown } from './Dropdown'
export { default as ThemeChange } from './ThemeChange'
