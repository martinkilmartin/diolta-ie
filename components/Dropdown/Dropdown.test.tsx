import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import Dropdown from './Dropdown'

test('Dropdown default', () => {
  const message = 'hello world'
  const dropdownRenderer = renderer.create(
    <Dropdown>
      <p>{message}</p>
    </Dropdown>
  )
  const dropdownAsJSON = dropdownRenderer.toJSON()
  expect(dropdownAsJSON).toMatchSnapshot()
  const dropdownRoot = dropdownRenderer.root
  expect(
    dropdownRoot.findByProps({ className: 'm-1 btn btn-secondary' }).children[0]
  ).toEqual('Change Theme')
  expect(dropdownRoot.findByType('p').children[0]).toEqual(message)
})
