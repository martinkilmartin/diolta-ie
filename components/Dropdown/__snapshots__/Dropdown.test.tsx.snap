// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown default 1`] = `
<div
  className="dropdown"
>
  <div
    className="m-1 btn btn-secondary"
    tabIndex={0}
  >
    Change Theme
  </div>
  <p>
    hello world
  </p>
</div>
`;
